lang/c++/impl/Compiler.cc

```diff
@@ -423,8 +423,12 @@ static NodePtr makeFixedNode(const Entity &e,
 static NodePtr makeArrayNode(const Entity &e, const Object &m,
                              SymbolTable &st, const string &ns) {
     auto it = findField(e, m, "items");
-    NodePtr node = NodePtr(std::make_shared<NodeArray>(
-        asSingleAttribute(makeNode(it->second, st, ns))));
+    std::shared_ptr<NodeArray> nodePointer = std::make_shared<NodeArray>(
+        asSingleAttribute(makeNode(it->second, st, ns)));
+    NodePtr node = NodePtr(nodePointer);
+    if (containsField(m, "element-id")) {
+        nodePointer->elementId_ = getStringField(e, m, "element-id");
```

Comment:
This also ends up being two lookups to the map between `containsField()` and `getStringField()`. I guess `avro` isn't too concerned about these lookups though, since they use this sub-optimal pattern throughout their functions. Could be a nice extension to add a function `std::optional<string> maybeGetStringField()` or something of the like in the future.